modules/caddyhttp/fileserver/matcher.go

```diff
@@ -139,6 +139,10 @@ func (m *MatchFile) Provision(_ caddy.Context) error {
 	if m.Root == "" {
 		m.Root = "{http.vars.root}"
 	}
+	// if list of files to try was omitted entirely, assume URL path
```

Comment:
Could we add the issue number to this comment? It's not immediately obvious why the placeholder would be used instead of simply `r.URL.Path` like we were before.
```suggestion
	// if list of files to try was omitted entirely, assume URL path
	// (use placeholder instead of r.URL.Path; see issue #4146)
```